maven/lib/dependabot/maven/update_checker/version_finder.rb

```diff
@@ -287,6 +283,14 @@ def central_repo_urls
 
           %w(http:// https://).map { |p| p + central_url_without_protocol }
         end
+
+        def auth_headers_finder
+          @auth_headers_finder ||= Utils::AuthHeadersFinder.new(credentials)
+        end
+
+        def auth_headers(maven_repo_url)
+          auth_headers_finder.auth_headers(maven_repo_url)
+        end
```

Comment:
We can do the same thing here right?

```suggestion
        def auth_headers
          @auth_headers ||= Utils::AuthHeadersFinder.new(credentials).auth_headers(maven_repo_url)
        end
```

Reply:
It actually turns out it was the tests not being thorough enough, since it only tested with a single defined maven_repository at the time. With multiple repositories defined, several requests will be made, with different maven_repo_urls. I have added tests and reverted the chang on the gradle version_finder.rb.

Reply:
Ah yes that makes sense 👍